lib/src/react16_suggestors/react_style_maps_updater.dart

```diff
@@ -0,0 +1,299 @@
+          // edits made.
+          if (affectedValues.isNotEmpty ||
+              isAVariable ||
+              isAFunction ||
+              isForCustomProps) {
+            yieldPatch(
+                cascade.offset,
+                cascade.offset,
+                getString(
+                    isAVariable: isAVariable,
+                    styleMapContainsAVariable: styleMapContainsAVariable,
+                    isAFunction: isAFunction,
+                    affectedValues: affectedValues,
+                    addExtraLine: sourceFile.getLine(cascade.offset) ==
+                        sourceFile.getLine(cascade.parent.offset),
+                    isForCustomProps: isForCustomProps));
+          }
+        }
+      }
+    }
+  }
+}
+
+String getString({
+  String styleMap,
+  List affectedValues = const [],
+  bool isAVariable = false,
+  bool styleMapContainsAVariable = false,
+  bool isAFunction = false,
+  bool addExtraLine = false,
+  bool isForCustomProps = false,
+}) {
+  String checkboxWithAffectedValues = '// [ ] Check this box upon manual '
+      'validation that this style map uses a valid value for the following '
+      'keys: ${affectedValues.join(', ')}.';
+
+  String variableCheckbox = '// [ ] Check this box upon manual validation '
+      'that this style map is receiving a value that is valid for the keys '
+      'that are simple string variables.';
+
+  String variableCheckboxWithAffectedValues = '// [ ] Check this box upon '
+      'manual validation that this style map is receiving a value that is '
+      'valid for the following keys: ${affectedValues.join(', ')}.';
+
+  String willBeRemovedSuffix = '//$willBeRemovedCommentSuffix';
+
+  String functionCheckbox = '// [ ] Check this box upon manual validation '
+      'that the method called to set the style prop does not return any '
+      'simple, unitless strings instead of nums.';
+
+  if (isForCustomProps) {
+    return '''
+    $variableCheckbox
+    $styleMapExample
+    $willBeRemovedSuffix
+    ''';
+  } else if ((isAVariable || styleMapContainsAVariable)) {
+    if (affectedValues.isNotEmpty) {
+      return '''
+      $variableCheckboxWithAffectedValues
+      $styleMapExample
+      $willBeRemovedSuffix
+      ''';
+    } else {
+      return '''
+      $variableCheckbox
+      $styleMapExample
+      $willBeRemovedSuffix
+      ''';
+    }
+  } else if (isAFunction) {
+    return '''
+       $functionCheckbox
+       $styleMapExample
+       $willBeRemovedSuffix
+       ''';
+  } else {
+    if (addExtraLine) {
+      return '''
+    
+      $checkboxWithAffectedValues
+      $styleMapExample
+      $willBeRemovedSuffix
+      ''';
+    } else {
+      return '''
+      $checkboxWithAffectedValues
+      $styleMapExample
+      $willBeRemovedSuffix
+      ''';
+    }
+  }
+}
+
+String cleanString(dynamic elementToClean) {
+  return elementToClean.toString().replaceAll("\'", "").replaceAll("\"", "");
```

Comment:
This could mess with interpolated strings, and the `\` parts of the string aren't being used because it's not a raw string. You'd need `r"\'"` or `"\\'`".

There's actually a utility to get string values from the AST node:
```dart
if (elementToClean is SimpleStringLiteral) {
  return elementToClean.value;
}
```
For other strings, I don't think we should try to handle them, but you can check for them with `is StringLiteral`.

Reply:
Similarly, you can also do `is IntLiteral || is DoubleLiteral` for numbers

Reply:
Ah, yep, that is more robust. Thank you!